src/dialog.mjs

```diff
@@ -105,59 +215,34 @@ const Dialog = ({ config, preferences }) => {
     // - Radio when an option has been selected.
     // - Checkbox with or without checked option, except the `acceptAllButton` case above.
     return values;
+  }
 
-  };
+  getPreferences() {
+    const preferences = Preferences(Config().get("prefix"));
 
-  /**
-   * Handle form submits.
-   */
-  const submitHandler = e => {
-    e.preventDefault();
+    return preferences;
+  }
 
-    // Get values based on the rules defined in `composeValues`.
-    const values = composeValues(tabList.getValues());
-    if (!values) {
-      return;
-    }
+  updatePreference(selectedCookies) {
+    this.preferences.store(selectedCookies);
+    this.events.dispatch("update", this.preferences.getAll());
+    this.domToggler.toggle(this.preferences);
+  }
 
-    // Dispatch values and hide the dialog.
-    events.dispatch('submit', values);
-    hide();
-  };
-
-  /**
-   * Update button label.
-   */
-  const updateButtonLabel = label => {
-    button.textContent = label;
-  };
-
-  return {
-    init() {
-      // Initialize tab list and append it to the form.
-      tabList.init();
-      form.insertBefore(tabList.element, form.firstElementChild);
-
-      // Attach submit listener to the form.
-      form.addEventListener('submit', preventingDefault(submitHandler));
-
-      // Update initial button label if `acceptAllButton` is set to true,
-      // and no cookie preferences have been stored. Also listen for changes
-      // in the form, and revert the button back to normal when an option has
-      // been selected.
-      if (ACCEPT_ALL_BUTTON) {
-        updateButtonLabel(config.get('labels.button.acceptAll'));
-        form.addEventListener('change', e => {
-          updateButtonLabel(config.get('labels.button.default'));
-        });
-      }
-    },
-    on: events.add,
-    show,
-    hide,
-    element: dialog,
-  };
+  initDomToggler() {
+    return DomToggler(Config());
+  }
+
+  show() {
```

Comment:
Just wondering, you will return a function from this method. Is it possible to make this method that exported function?
so: 
```
show() {
this.shadowRoot.querySelector('.cookie-consent').setAttribute("aria-hidden", "false");
}
```